### ISSUE TITLE
Message: "***It seems that Metis did not free all of its memory! Report this."

### ISSUE THREAD

**Issue by cponder** [KarypisLab/METIS#51]
I ran this command on a weather grid from NCAR at 5km resolution:
```
gpmetis -minconn -contig -niter=200 x1.*.graph.info 1792
```
This is the output it gave:
```
******************************************************************************
METIS 5.2 Copyright 1998-16, Regents of the University of Minnesota
 (HEAD: , Built on: Nov 30 2022, 10:22:25)
 size of idx_t: 32bits, real_t: 32bits, idx_t *: 64bits

Graph Information -----------------------------------------------------------
 Name: x1.5898242.graph.info, #Vertices: 5898242, #Edges: 17694720, #Parts: 1792

Options ---------------------------------------------------------------------
 ptype=kway, objtype=cut, ctype=shem, rtype=greedy, iptype=metisrb
 dbglvl=0, ufactor=1.030, no2hop=NO, minconn=YES, contig=YES
 ondisk=NO, nooutput=NO
 seed=-1, niparts--1, niter=200, ncuts=1

Direct k-way Partitioning ---------------------------------------------------
free(): invalid pointer
free(): invalid pointer
***It seems that Metis did not free all of its memory! Report this.

***Metis returned with an error.
Could not find pointer 0x7fad5e41f010 in mcore

Aborted (core dumped)
```
This is the only case I've seen of ```gpmetis``` failing. Does this happen if I exhaust the RAM memory?

**Comment by cponder** [comment]
This version didn't have the problem:
http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/metis-5.1.0.tar.gz

**Comment by karypis** [comment]
Can you try to reproduce the error by trying a few different seeds for the random number generator (it is one of the optional command line parameters)?

Can you share the graph?

**Comment by cponder** [comment]
Here's a data-set where I see the breakage:
https://www2.mmm.ucar.edu/projects/mpas/benchmark/v7.0/MPAS-A_benchmark_10km_v7.0.tar.gz

**Comment by cponder** [comment]
I'm building under Ubuntu 22.04 using the GCC 11.3.0 compilers.

**Comment by cponder** [comment]
I tried these few, which all failed:
```
gpmetis -seed 240 -minconn -contig -niter=200 x1.*.graph.info  256
gpmetis -seed 8771326 -minconn -contig -niter=200 x1.*.graph.info  256
gpmetis -seed 38771326 -minconn -contig -niter=200 x1.*.graph.info  256
```

**Comment by karypis** [comment]
I tried on a fresh build of Metis on my m1 based macbook pro and everything looks fine.

<img width="867" alt="Screen Shot 2022-12-03 at 6 00 20 PM" src="https://user-images.githubusercontent.com/1174919/205467411-03430bb5-dfcd-4ba5-8567-9caa09e12118.png">

<img width="945" alt="Screen Shot 2022-12-03 at 5 58 59 PM" src="https://user-images.githubusercontent.com/1174919/205467416-5de0212f-3e79-4896-a72f-d35e4c95c0e8.png">

**Comment by cponder** [comment]
I'm running inside a container. Maybe it's causing some interference with the memory-allocation.
Are there some print-statements that I can insert, that would explain where the failure is happening?

**Comment by cponder** [comment]
Here's a discrepancy:
```
size of idx_t: 32bits, real_t: 32bits, idx_t *: 64bits
```
When you run it, the `idx_t` says it's `64bits` not `32bits`.

**Comment by karypis** [comment]
I use the latest from the master branch.

**Comment by cponder** [comment]
Yes this did work. I had to also use the master-branch version of the GKlib.
Can you please make new releases of these?

**Comment by cponder** [comment]
Also, it still says `idx_t: 32bits` here:
```
 size of idx_t: 32bits, real_t: 32bits, idx_t *: 64bits
```
Can you tell me why it's doing this?

**Comment by karypis** [comment]
My built was using 64 bit for idx_t ('make config i64=1'). It does not make a difference.

**Comment by cponder** [comment]
Yeah I get that, the idx_t: 32bits ran ok for me too.
We're seeing memory failures with a different app, though, and I wanted to make sure there wasn't some issue with our container constricting some type to 32-bit.

**Comment by cponder** [comment]
The latest tag ```[v5.1.1-DistDGL-v0.5``` is from over 2 years ago, and (incorrectly) reads ```METIS 5,2``` when I run it.
Can you please tag a 5.2 snapshot?
